pkg/kubelet/volumemanager/populator/desired_state_of_world_populator.go

```diff
@@ -245,6 +245,10 @@ func (dswp *desiredStateOfWorldPopulator) findAndRemoveDeletedPods() {
 			continue
 		}
 
+		if !dswp.actualStateOfWorld.VolumeExists(volumeToMount.VolumeName) && podExists {
```

Comment:
What happens when we quickly add a pod and then delete it. The code will wait for volume be attached before it can be mounted and hence it will not be present in ASW. But the entry will not be removed from DSW even when pod is deleted because of this check right?

Reply:
because we have an extra check on podExists here, so the entry should be removed. When pod is added and quickly deleted before reconciler sets up mount for it, kubelet should be able to delete it without problem, and podManager will not have this pod anymore, so podExists will become false.